spec/unit/resource/api300/synergy/interconnect_spec.rb

```diff
@@ -25,6 +25,56 @@
     end
   end
 
+  describe '#retrieve!' do
+    before :each do
+      resp = FakeResponse.new(members: [
+        { name: 'name1', uri: 'uri1', serialNumber: 'sn1' },
+        { name: 'name2', uri: 'uri2', serialNumber: 'sn2' }
+      ])
+      allow(@client_300).to receive(:rest_get).with(described_class::BASE_URI).and_return(resp)
+    end
+
+    it 'retrieves by name' do
+      expect(described_class.new(@client_300, name: 'name1').retrieve!).to be true
+      expect(described_class.new(@client_300, name: 'fake').retrieve!).to be false
+    end
+
+    it 'retrieves by uri' do
+      expect(described_class.new(@client_300, uri: 'uri1').retrieve!).to be true
+      expect(described_class.new(@client_300, uri: 'fake').retrieve!).to be false
+    end
+
+    it 'retrieves by serialNumber' do
+      expect(described_class.new(@client_300, serialNumber: 'sn1').retrieve!).to be true
+      expect(described_class.new(@client_300, serialNumber: 'fake').retrieve!).to be false
+    end
+  end
+
+  describe '#exists?' do
+    before :each do
+      resp = FakeResponse.new(members: [
+        { name: 'name1', uri: 'uri1', serialNumber: 'sn1' },
+        { name: 'name2', uri: 'uri2', serialNumber: 'sn2' }
+      ])
+      allow(@client_300).to receive(:rest_get).with(described_class::BASE_URI).and_return(resp)
+    end
+
+    it 'finds it by name' do
+      expect(described_class.new(@client_300, name: 'name1').exists?).to be true
+      expect(described_class.new(@client_300, name: 'fake').exists?).to be false
+    end
+
+    it 'finds it by uri' do
+      expect(described_class.new(@client_300, uri: 'uri1').exists?).to be true
+      expect(described_class.new(@client_300, uri: 'fake').exists?).to be false
+    end
+
+    it 'finds it by serialNumber' do
+      expect(described_class.new(@client_300, serialNumber: 'sn1').exists?).to be true
```

Comment:
Just a tip, no need to change.

But it is triple redundant here:
1.  `.to be` acts the same as `.to eq(true)`, so no need to write `to be true` nor `to eq(true)`
2.  Any method ended with `?` is automatically transformed into a matcher by RSpec
  **E.G.** .zero? automatically gains a matcher be_zero, empty?... be_empty
   So no need to do `expect(clause.method?).to be` just use `expect(clause).to be_method`
3. At last, RSpec has one exclusive matcher for **#exist?** or **#exists?** named `exist` (Although usually only checking if it is truthy with **.to be** is pretty much enough for most cases)